### PR TITLE
Login Notice: Add Rich Text Editor

### PR DESCRIPTION
**Pull request by SammyIsConfused**
#660 
This PR adds a react-draft-wysiwyg rich text editor to the login notice configurator, which can support rich text, links, and images (currently only via URL, image upload will be a separate PR)

image
image
image

### CODE REVIEW

File: Source/Plugins/Core/com.equella.core/js/tsrc/components/RichTextEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+            case "left":
+              textAlign = "left";
+              break;
+            case "right":
+              textAlign = "right";
+              break;
+          }
+        }
+        return (
+          '<imagealign style="display:flex; justify-content:' +
```
Comment (ChristianMurphy) [inline Source/Plugins/Core/com.equella.core/js/tsrc/components/RichTextEditor.tsx]: :art: lines 51-63 be a bit easier to read as a template literal.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

File: Source/Plugins/Core/com.equella.core/js/tsrc/components/RichTextEditor.tsx [hunk 2]
```diff
@@ -0,0 +1,128 @@
+import { Editor } from "react-draft-wysiwyg";
+import { ContentState, convertToRaw, EditorState } from "draft-js";
+import * as React from "react";
+
+let draftjsToHtml: Function = require("draftjs-to-html");
+let htmlToDraft: Function = require("html-to-draftjs").default;
+
+require("react-draft-wysiwyg/dist/react-draft-wysiwyg.css");
```
Comment (ChristianMurphy) [inline Source/Plugins/Core/com.equella.core/js/tsrc/components/RichTextEditor.tsx]: `import 'react-draft-wysiwyg/dist/react-draft-wysiwyg.css'`

File: Source/Plugins/Core/com.equella.core/resources/web/css/react-draft-wysiwyg.css [hunk 3]
```diff
@@ -0,0 +1,1041 @@
...
+  padding-left: 5px;
+}
+.rdw-editor-main pre {
+  background: #f1f1f1;
+  border-radius: 3px;
+  padding: 1px 10px;
+} /**
+ * Draft v0.9.1
+ *
+ * Copyright (c) 2013-present, Facebook, Inc.
```
Comment (ChristianMurphy) [inline Source/Plugins/Core/com.equella.core/resources/web/css/react-draft-wysiwyg.css]: Is this duplicating the import from RichTextEditor line 8?

**Comment by PenghaiZhang** [comment]
Hi Sam, just confirm that all of us use axios to perform requests and handle errors etc.  Am I right?

File: Source/Plugins/Core/com.equella.core/resources/web/css/react-draft-wysiwyg.css (hunk 3 above)
Comment (SammyIsConfused) [reply Source/Plugins/Core/com.equella.core/resources/web/css/react-draft-wysiwyg.css]: Yes it is! This is left over from before the migration to parcel from browserify. I'll get rid of it.

File: Source/Plugins/Core/com.equella.core/js/tsrc/components/RichTextEditor.tsx (hunk 1 above)
Comment (SammyIsConfused) [reply Source/Plugins/Core/com.equella.core/js/tsrc/components/RichTextEditor.tsx]: Great find! Yeah, that's so much easier to understand.

File: Source/Plugins/Core/com.equella.core/js/tsrc/components/RichTextEditor.tsx (hunk 2 above)
Comment (SammyIsConfused) [reply Source/Plugins/Core/com.equella.core/js/tsrc/components/RichTextEditor.tsx]: Doesn't work, as css files need to be packaged and aren't modules in themselves. This must be require for Parcel to pick it up.

**Comment by SammyIsConfused** [comment]
> Hi Sam, just confirm that all of us use axios to perform requests and handle errors etc. Am I right?

That is correct, Axios is what has been used for HTTP requests and responses in recent development work.

File: Source/Plugins/Core/com.equella.core/js/tsrc/components/RichTextEditor.tsx (hunk 2 above)
Comment (doolse) [reply Source/Plugins/Core/com.equella.core/js/tsrc/components/RichTextEditor.tsx]: @ChristianMurphy is right in this case, import syntax will work when it's a css file.

Comment (SammyIsConfused) [reply Source/Plugins/Core/com.equella.core/js/tsrc/components/RichTextEditor.tsx]: just tried it again, yeah sorry I forgot to remove the parentheses. That fixed it.

File: Source/Plugins/Core/com.equella.core/src/com/tle/web/api/loginnotice/PreLoginNoticeResource.java [hunk 4]
```diff
@@ -19,16 +19,18 @@
 import io.swagger.annotations.Api;
 import javax.ws.rs.*;
 import javax.ws.rs.core.Response;
+import org.mortbay.jetty.MimeTypes;
```
Comment (doolse) [inline Source/Plugins/Core/com.equella.core/src/com/tle/web/api/loginnotice/PreLoginNoticeResource.java]: We don't actually use jetty on the server side, so it's probably better to use some other mimetype constants.`javax.ws.rs.core.MediaType`

**Comment by SammyIsConfused** [comment]
Verbally approved by @doolse.

**Pull request merged**